src/app/Main/Body/QueryPage/Drawer/Filter/FilterSetting-Dialog.component.tsx

```diff
@@ -0,0 +1,155 @@
+/** @jsx jsx */
+import { css, jsx } from "@emotion/core";
+import { useState } from "react";
+import _ from "lodash";
+import {
+    Button,
+    Dialog,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    FormGroup,
+} from "@material-ui/core";
+import DoneIcon from "@material-ui/icons/Done";
+import { Link } from "react-router-dom";
+import {
+    onPrimaryColor,
+    primaryColor,
+    secondaryColor,
+} from "../../../../../Shared/Style/Style-MainTheme.component";
+import { CheckboxComponent } from "../../../../../Shared/Checkbox.component";
+
+const buttonStyle = css`
+    width: 100%;
+    height: 1.5rem;
+    margin-top: 0.5em;
+    background-color: ${primaryColor};
+    color: ${onPrimaryColor};
+    &:hover {
+        background-color: ${primaryColor};
+        color: ${secondaryColor};
+    }
+`;
+const linkStyle = css`
+    color: ${primaryColor};
+    &:hover {
+        color: ${secondaryColor};
+    }
+`;
+
+function filterCheckboxes(
+    mainFilters: string[],
+    displayedFilter: string[],
+    handleChangeCheckbox: (name: string, checked: boolean) => void
+): JSX.Element[] {
+    const checkboxes: JSX.Element[] = [];
+    mainFilters.forEach((mainFilter) => {
+        const isChecked: boolean = _.includes(displayedFilter, mainFilter);
+        checkboxes.push(
+            <CheckboxComponent
+                onCheckboxChange={handleChangeCheckbox}
+                checked={isChecked}
+                key={`checkbox-${mainFilter}`}
+                name={mainFilter}
+                label={mainFilter}
+            />
+        );
+    });
+    return checkboxes;
+}
+
+export function FilterSettingDialogComponent(props: {
+    isOpen: boolean;
+    onClickOpen: () => void;
+    onClickClose: () => void;
+    mainFilters: string[];
+    displayedFilter: string[];
+    onSubmitClick: (filterToDisplay: string[]) => void;
+}): JSX.Element {
+    const [filterToDisplay, setFilterToDisplay] = useState<string[]>(
+        props.displayedFilter
+    );
+    const handleClickOpen = (): void => props.onClickOpen();
+    const handleClickClose = (): void => props.onClickClose();
+
+    const handleClickCancel = (): void => {
+        setFilterToDisplay(props.displayedFilter);
+        handleClickClose();
+    }
+
+    const handleClickSubmit = (): void => {
+        props.onSubmitClick(filterToDisplay);
+        handleClickClose();

```

Comment:
Reject: emitting two events here is not a good idea, just emit onSubmitClick and handle the closing in the parent component.

Reply:
Done. Commit "Move Closing form dialog to parent component"